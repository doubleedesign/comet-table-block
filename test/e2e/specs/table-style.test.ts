/**
 * WordPress dependencies
 */
import {
	getEditedPostContent,
	createNewPost,
	transformBlockTo,
	clickBlockToolbarButton,
	clickButton,
} from '@wordpress/e2e-test-utils';

/**
 * Internal dependencies
 */
import {
	createNewFlexibleTableBlock,
	createNewCoreTableBlock,
	coreTableCellSelector,
	flexibleTableCellSelector,
	flexibleTableCaptionSelector,
	clickButtonWithAriaLabel,
	clickButtonWithText,
	inputValueFromLabel,
	inputValueFromAriaLabel,
	inputValueFromLabelledBy,
	clickToggleControlWithText,
	getWpVersion,
	openSidebar,
	openSidebarPanelWithTitle,
} from '../helper';

/** @type {import('puppeteer').Page} */
const page = global.page;

describe( 'Style', () => {
	beforeEach( async () => {
		await createNewPost();
	} );

	it( 'caption settings should be applied', async () => {
		await createNewFlexibleTableBlock();
		await page.$$( flexibleTableCaptionSelector );
		await page.focus( flexibleTableCaptionSelector );
		await page.keyboard.type( 'Flexible Table Block' );
		await openSidebar();
		await openSidebarPanelWithTitle( 'Caption Settings' );
		await inputValueFromLabel( 'Caption Font Size', '20px' );
		await inputValueFromLabel( 'Caption Line Height', '2' );
		await clickButtonWithAriaLabel( '.ftb-padding-control__header-control', 'Unlink Sides' );
		await inputValueFromAriaLabel( '.ftb-padding-control__input-controls', 'Top', '1' );
		await inputValueFromAriaLabel( '.ftb-padding-control__input-controls', 'Right', '2' );
		await page.keyboard.press( 'Tab' );
		await page.keyboard.press( 'ArrowDown' );
		await inputValueFromAriaLabel( '.ftb-padding-control__input-controls', 'Bottom', '3' );
		await page.keyboard.press( 'Tab' );
		await page.keyboard.press( 'ArrowDown' );
		await page.keyboard.press( 'ArrowDown' );
		await inputValueFromAriaLabel( '.ftb-padding-control__input-controls', 'Left', '4' );
		await page.keyboard.press( 'Tab' );
		await page.keyboard.press( 'ArrowDown' );
		await page.keyboard.press( 'ArrowDown' );
		await clickButtonWithText(
			'//*[@aria-labelledby="flexible-table-block-caption-side-heading"]',
			'Top'
		);
		await clickButtonWithAriaLabel( '.edit-post-sidebar', 'Align center' );

		expect( await getEditedPostContent() ).toMatchSnapshot();
	} );
} );
