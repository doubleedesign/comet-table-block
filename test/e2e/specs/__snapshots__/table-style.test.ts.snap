// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Styles caption styles should be applied 1`] = `
"<!-- wp:flexible-table-block/table {\\"captionSide\\":\\"top\\"} -->
<figure class=\\"wp-block-flexible-table-block-table\\"><figcaption style=\\"font-size:20px;line-height:2;padding:1px 2% 3em 4rem;text-align:center\\">Flexible Table Block</figcaption><table class=\\"has-fixed-layout\\"><tbody><tr><td></td><td></td><td></td></tr><tr><td></td><td></td><td></td></tr><tr><td></td><td></td><td></td></tr></tbody></table></figure>
<!-- /wp:flexible-table-block/table -->"
`;

exports[`Styles cell styles should be applied 1`] = `
"<!-- wp:flexible-table-block/table -->
<figure class=\\"wp-block-flexible-table-block-table\\"><table class=\\"has-fixed-layout\\"><tbody><tr><th class=\\"custom\\" id=\\"id\\" headers=\\"headers\\" scope=\\"row\\" style=\\"font-size:20px;line-height:2;width:100px;color:#111111;background-color:#333333;padding:1px 2% 3em 4rem;border-radius:1px 2em 3rem 4rem;border-width:1px 2em 3rem 4rem;border-style:solid dotted dashed double;border-color:#111111 #222222 #333333 #444444;text-align:center;vertical-align:middle\\"></th><td></td><td></td></tr><tr><td></td><td></td><td></td></tr><tr><td></td><td></td><td></td></tr></tbody></table></figure>
<!-- /wp:flexible-table-block/table -->"
`;

exports[`Styles cell styles should be applied to multiple cells 1`] = `
"<!-- wp:flexible-table-block/table -->
<figure class=\\"wp-block-flexible-table-block-table\\"><table class=\\"has-fixed-layout\\"><thead><tr><th></th><th></th><th class=\\"custom\\" id=\\"id\\" headers=\\"headers\\" scope=\\"row\\" style=\\"font-size:20px;line-height:2;width:100px;color:#111111;background-color:#333333;padding:1px 2% 3em 4rem;border-radius:1px 2em 3rem 4rem;border-width:1px 2em 3rem 4rem;border-style:solid dotted dashed double;border-color:#111111 #222222 #333333 #444444;text-align:center;vertical-align:middle\\"></th></tr></thead><tbody><tr><td></td><td></td><th class=\\"custom\\" id=\\"id\\" headers=\\"headers\\" scope=\\"row\\" style=\\"font-size:20px;line-height:2;width:100px;color:#111111;background-color:#333333;padding:1px 2% 3em 4rem;border-radius:1px 2em 3rem 4rem;border-width:1px 2em 3rem 4rem;border-style:solid dotted dashed double;border-color:#111111 #222222 #333333 #444444;text-align:center;vertical-align:middle\\"></th></tr><tr><td></td><td></td><th class=\\"custom\\" id=\\"id\\" headers=\\"headers\\" scope=\\"row\\" style=\\"font-size:20px;line-height:2;width:100px;color:#111111;background-color:#333333;padding:1px 2% 3em 4rem;border-radius:1px 2em 3rem 4rem;border-width:1px 2em 3rem 4rem;border-style:solid dotted dashed double;border-color:#111111 #222222 #333333 #444444;text-align:center;vertical-align:middle\\"></th></tr><tr><td></td><td></td><th class=\\"custom\\" id=\\"id\\" headers=\\"headers\\" scope=\\"row\\" style=\\"font-size:20px;line-height:2;width:100px;color:#111111;background-color:#333333;padding:1px 2% 3em 4rem;border-radius:1px 2em 3rem 4rem;border-width:1px 2em 3rem 4rem;border-style:solid dotted dashed double;border-color:#111111 #222222 #333333 #444444;text-align:center;vertical-align:middle\\"></th></tr></tbody><tfoot><tr><td></td><td></td><th class=\\"custom\\" id=\\"id\\" headers=\\"headers\\" scope=\\"row\\" style=\\"font-size:20px;line-height:2;width:100px;color:#111111;background-color:#333333;padding:1px 2% 3em 4rem;border-radius:1px 2em 3rem 4rem;border-width:1px 2em 3rem 4rem;border-style:solid dotted dashed double;border-color:#111111 #222222 #333333 #444444;text-align:center;vertical-align:middle\\"></th></tr></tfoot></table></figure>
<!-- /wp:flexible-table-block/table -->"
`;

exports[`Styles table styles should be applied 1`] = `
"<!-- wp:flexible-table-block/table {\\"hasFixedLayout\\":false,\\"isScrollOnPc\\":true,\\"isScrollOnMobile\\":true,\\"isStackedOnMobile\\":true,\\"sticky\\":\\"first-column\\"} -->
<figure class=\\"wp-block-flexible-table-block-table is-scroll-on-pc is-scroll-on-mobile\\"><table class=\\"is-stacked-on-mobile is-sticky-first-column\\" style=\\"width:500px;max-width:600%;min-width:400em;padding:1px 2% 3em 4rem;border-radius:1px 2em 3rem 4rem;border-width:1px 2em 3rem 4rem;border-style:solid dotted dashed double;border-color:#111111 #222222 #333333 #444444;border-collapse:separate;border-spacing:10px 5em\\"><tbody><tr><td></td><td></td><td></td></tr><tr><td></td><td></td><td></td></tr><tr><td></td><td></td><td></td></tr></tbody></table></figure>
<!-- /wp:flexible-table-block/table -->"
`;
